src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProjectProvider.cs

```diff
@@ -62,32 +67,30 @@ public bool TryCreateNuGetProject(EnvDTE.Project dteProject, ProjectSystemProvid
             {
                 return false;
             }
-            var restoreProjectStyle = string.Empty;
-            var targetFramework = string.Empty;
-            var targetFrameworks = string.Empty;
-
-            var hasRestoreProjectStyle = context
-                .MSBuildProperties
-                .TryGetValue(ProjectSystemProviderContext.RestoreProjectStyle, out restoreProjectStyle) 
-                && !string.IsNullOrEmpty(restoreProjectStyle);
-
-            var hasTargetFramework = context
-                .MSBuildProperties
-                .TryGetValue(ProjectSystemProviderContext.TargetFramework, out targetFramework)
-                && !string.IsNullOrEmpty(targetFramework);
-
-            var hasTargetFrameworks = context
-                .MSBuildProperties
-                .TryGetValue(ProjectSystemProviderContext.TargetFrameworks, out targetFrameworks)
-                && !string.IsNullOrEmpty(targetFrameworks);
-
-            // check for RestoreProjectStyle property is set and if set to PackageReference then return false
-            if (hasRestoreProjectStyle && !restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase))
+
+            // Check if the project is not CPS capable or if it is CPS capable then it does not have TargetFramework(s), if so then return false
+            if (!hierarchy.IsCapabilityMatch("CPS"))
             {
                 return false;
             }
-            // Check if the project is not CPS capable or if it is CPS capable then it does not have TargetFramework(s), if so then return false
-            else if (!(hierarchy.IsCapabilityMatch("CPS") && (hasTargetFramework || hasTargetFrameworks)))
+
+            var buildPropertyStorage = hierarchy as IVsBuildPropertyStorage;
+
+            // read MSBuild property RestoreProjectStyle, TargetFramework, and TargetFrameworks
+            var restoreProjectStyle = VsHierarchyUtility.GetMSBuildProperty(buildPropertyStorage, RestoreProjectStyle);
+
+            var targetFramework = VsHierarchyUtility.GetMSBuildProperty(buildPropertyStorage, TargetFramework);
+
+            var targetFrameworks = VsHierarchyUtility.GetMSBuildProperty(buildPropertyStorage, TargetFrameworks);
+
+            // check for RestoreProjectStyle property is set and if not set to PackageReference then return false
+            if (!(string.IsNullOrEmpty(restoreProjectStyle) || 
```

Comment:
This really looks like it changed to the opposite unless I'm missing something.

``Before: !restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase)`` -> return false

``After: restoreProjectStyle.Equals(ProjectStyle.PackageReference.ToString(), StringComparison.OrdinalIgnoreCase)`` -> return false

https://github.com/NuGet/NuGet.Client/pull/1160/files#diff-ee3999f1655ce6b9b9e0ed7454096e2dL85